prusti-viper/src/encoder/procedure_encoder.rs

```diff
@@ -2053,7 +2053,13 @@ impl<'p, 'v: 'p, 'tcx: 'v> ProcedureEncoder<'p, 'v, 'tcx> {
                                     )?);
                                 }
 
-                                _ => unreachable!()
+                                _ => {
+                                    cleanup(&self);
+                                    return Err(SpannedEncodingError::unsupported(
+                                        format!("only calls to closures are supported. The term is a {:?}, not a closure.", cl_type.kind()),
```

Comment:
Sorry I just realised this now, but this error message is a bit misleading – it sounds as if we didn't support the most basic case of calls to regular functions.

Reply:
True, I didn't think about that interpretation.